g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/service/SingleComponentServiceImpl.java

```diff
@@ -213,52 +217,167 @@ private boolean synchkey2VipL10n(RecordModel record,  String key, String srcValu
         
         return true;
 	}
-	
-	
-	
-    /**
-     * sync the local source content to internal l10n or i18n
-     */
+
+
+
+	/**
+	 * sync the local source content to internal l10n or i18n
+	 */
 	@Override
 	public boolean synchComponentFile2Internal(RecordModel record) {
 
-      ComponentSourceModel model =  getSourceComponentFile(record);
-      if (model != null) {
-		  for (Entry<String, Object> entry : model.getMessages().entrySet()) {
-			  try {
-				  boolean result = synchkey2VipL10n(record, entry.getKey(), (String) entry.getValue(), null, null);
-				  if (!result) {
-					  result = synchkey2VipI18n(record, entry.getKey(), (String) entry.getValue(), null, null);
-					  if (!result) {
-						  try {
-							  if (record.getStatus() < 5) {
-								  record.setStatus(record.getStatus() + 1);
-								  TaskSysnQueues.SendComponentTasks.put(record);
-							  }
-						  } catch (InterruptedException e) {
-
-							  Thread.currentThread().interrupt();
-
-							  logger.info(e.getMessage(), e);
-						  }
-						  return result;
-					  }
-				  }
-			  } catch (UnsupportedEncodingException e) {
-
-				  logger.error(e.getMessage(), e);
-				  return false;
-			  }
-		  }
-	  }
-		return true;
+		ComponentSourceModel model =  getSourceComponentFile(record);
+		if (this.configs.isSyncBatchEnable()){
+			return syncBatchSource(record, model);
+		}else {
+		  return syncSingleSource(record, model);
```

Comment:
if syncBatchSource is better, can syncSingleSource be removed?

Reply:
Because of the GCC not stable now，We still need use other environment source collection that use the sync single source. we can't remove it now. I think that we can remove it after all source collection use GCC.